data/functions/itemized.sql

```diff
@@ -1,3 +1,111 @@
+BEGIN
+    FOR cycle IN start_year..end_year BY 2 LOOP
+        child_table_name = format('ofec_sched_%s_%s_%s_tmp', schedule, cycle - 1, cycle);
+        EXECUTE format('CREATE TABLE %I (
+            CHECK ( two_year_transaction_period in (%s, %s) )
+        ) INHERITS (%I)', child_table_name, cycle - 1, cycle, master_table_name);
+
+    END LOOP;
+END
+$$ LANGUAGE plpgsql;
+
+
+-- Performs the final steps needed to setup all of the child tables for
+-- schedule A.
+CREATE OR REPLACE FUNCTION finalize_itemized_schedule_a_tables(start_year NUMERIC, end_year NUMERIC) RETURNS VOID AS $$
+DECLARE
+    child_table_name TEXT;
+    child_table_root TEXT;
+BEGIN
+    FOR cycle in start_year..end_year BY 2 LOOP
+        child_table_root = format('ofec_sched_a_%s_%s', cycle - 1, cycle);
+        child_table_name = format('ofec_sched_a_%s_%s_tmp', cycle - 1, cycle);
+
+        -- Create indexes.
+        EXECUTE format('CREATE UNIQUE INDEX idx_%s_sub_id_tmp ON %I (sub_id)', child_table_root, child_table_name);
+        EXECUTE format('ALTER TABLE %I ADD PRIMARY KEY USING INDEX idx_%s_sub_id_tmp', child_table_name, child_table_root);
+        EXECUTE format('CREATE INDEX idx_%s_rpt_yr_tmp ON %I (rpt_yr)', child_table_root, child_table_name);
+        EXECUTE format('CREATE INDEX idx_%s_pg_date_tmp ON %I (pg_date)', child_table_root, child_table_name);
+        EXECUTE format('CREATE INDEX idx_%s_entity_tp_tmp ON %I (entity_tp)', child_table_root, child_table_name);
+        EXECUTE format('CREATE INDEX idx_%s_image_num_tmp ON %I (image_num)', child_table_root, child_table_name);
+        EXECUTE format('CREATE INDEX idx_%s_contbr_st_tmp ON %I (contbr_st)', child_table_root, child_table_name);
+        EXECUTE format('CREATE INDEX idx_%s_contbr_city_tmp ON %I (contbr_city)', child_table_root, child_table_name);
+        EXECUTE format('CREATE INDEX idx_%s_is_individual_tmp ON %I (is_individual)', child_table_root, child_table_name);
+        EXECUTE format('CREATE INDEX idx_%s_clean_contbr_id_tmp ON %I (clean_contbr_id)', child_table_root, child_table_name);
+        EXECUTE format('CREATE INDEX idx_%s_two_year_transaction_period_tmp ON %I (two_year_transaction_period)', child_table_root, child_table_name);
+        EXECUTE format('CREATE INDEX idx_%s_sub_id_amount_tmp ON %I (contb_receipt_amt, sub_id)', child_table_root, child_table_name);
+        EXECUTE format('CREATE INDEX idx_%s_sub_id_date_tmp ON %I (contb_receipt_dt, sub_id)', child_table_root, child_table_name);
+        EXECUTE format('CREATE INDEX idx_%s_sub_id_cmte_id_tmp ON %I (cmte_id, sub_id)', child_table_root, child_table_name);
+        EXECUTE format('CREATE INDEX idx_%s_sub_id_cmte_id_amount_tmp ON %I (cmte_id, contb_receipt_amt, sub_id)', child_table_root, child_table_name);
+        EXECUTE format('CREATE INDEX idx_%s_sub_id_cmte_id_date_tmp ON %I (cmte_id, contb_receipt_dt, sub_id)', child_table_root, child_table_name);
```

Comment:
Disregard, we're going to follow up with this separately and not as a part of this PR!